### PR TITLE
sortDataLoggerFiles.py: fixed duplicate file identification

### PR DESCRIPTION
**Pull request by PetrilloAtWork**
**TL;DR:** `sortDataLoggerFiles.py` couldn't detect duplicate files any more. This patch fixes it.

A recent DAQ change in the output file name pattern introduced two numbers (an `art` and a `EventBuilder` index) in place of a single one (data logger, `dl`). To support both formats, the number that used to represent the data logger in `sortDataLoggerFiles.py` was replaced by an object that could represent either a data logger or a builder+thread pair.

This quantity (formerly number, now object) was also directly used as a unique key to identify duplicate files. Python then implicitly used the address of an object as component of that key instead of the value of the number (or of the object). Duplicate files would get different objects (with different addresses) with the same value, which would make for different keys. With this fix, the value of the object is explicitly used instead of the address.

Reviewers: @mvicenzi as DAQ convener witnessing the DAQ change.

### CODE REVIEW

**Comment by mmrosenberg** [comment]
trigger build

**Comment by mmrosenberg** [comment]
trigger build LArSoft/lar*@LARSOFT_SUITE_v09_90_00

**Comment by mmrosenberg** [comment]
trigger build LArSoft/lar*@LARSOFT_SUITE_v09_90_00
